x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -498,27 +500,36 @@ public void invalidateApiKeyForApiKeyName(String apiKeyName, ActionListener<Inva
     }
 
     private void findActiveApiKeysForUserAndRealm(String userName, String realmName, ActionListener<Collection<String>> listener) {
+        findApiKeysForUserAndRealm(userName, realmName, true, ActionListener.wrap(apiKeyInfos -> {
+            listener.onResponse(apiKeyInfos.stream().map(o -> o.getId()).collect(Collectors.toList()));
+        }, listener::onFailure));
+    }
+
+    private void findApiKeysForUserAndRealm(String userName, String realmName, boolean onlyActiveApiKeys,
```

Comment:
We handle the `invalidated` keys explicitly in this but not the expired ones. This is a little unrelated to this specific PR but wouldn't it make sense for `findActiveApiKeysForUserAndRealm` to get the api keys that are neither explicitly invalidated nor expired ? In other words, do we need to invalidate the API keys that have expired?

Reply:
Yes, I agree that there is no value in invalidating expired keys